src/module.cpp

```diff
@@ -471,6 +471,108 @@ void Module::AddTypeDef(const std::string &name, const Type *type, SourcePos pos
     symbolTable->AddType(name.c_str(), type, pos);
 }
 
+// Construct ConstExpr as initializer for varying const value from given expression list if possible
+template <class T>
+Expr *lCreateConstExpr(ExprList *exprList, const AtomicType::BasicType basicType, const Type *type,
+                       const std::string &name, SourcePos pos) {
+    const int N = g->target->getVectorWidth();
+    bool canConstructConstExpr = true;
+    T vals;
+    if constexpr (std::is_same_v<T, std::vector<llvm::APFloat>>) {
+        for (size_t j = 0; j < N; j++) {
+            switch (basicType) {
+            case AtomicType::TYPE_FLOAT16:
+                vals.push_back(llvm::APFloat::getZero(llvm::APFloat::IEEEhalf()));
```

Comment:
Are there any reasons why not to use `std::fill` here instead and remove `for` loop?
```
std::fill(vals.begin(), vals.end(), llvm::APFloat::getZero(llvm::APFloat::IEEEhalf()));
```

Reply:
I have removed the `for` loop using `vals.resize(N, llvm::APFloat::getZero(llvm::APFloat::IEEEhalf()))`.